### ISSUE TITLE
Provide shared libraries with fltk-config

### ISSUE THREAD

**Issue by dvzrv** [fltk/fltk#22]
The fltk-config tool should provide the path to `libfltk.so` to be able to build against the shared libraries.
Alternatively it would be very benificial to provide pkgconfig integration.

**Comment by Albrecht-S** [comment]
This is a known issue and should be checked, thanks.

We're also considering to provide pkgconfig files, but this is not yet in development. Patches and PR's welcome ;-)

**Comment by KBentley57** [comment]
I'm building fltk as an external project with cmake, I was able to work around this with until the issue is resolved using add_custom_command, where FLTK is built with ExternalProject_Add.  

```
ExternalProject_ADD(FLTK
	URL ${CMAKE_CURRENT_SOURCE_DIR}/fltk-1.3.5-source.tar.xz
	URL_MD5 35f276c93e7d724efd2f63b43f13160e  
	PREFIX ${CMAKE_CURRENT_BINARY_DIR}/fltk
	INSTALL_DIR ${SUPPORTPACKAGES_INSTALL_PREFIX}
	CMAKE_ARGS -DCMAKE_PREFIX_PATH=<INSTALL_DIR>
			   -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
			   -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}			   
			   -DOPTION_BUILD_SHARED_LIBS=TRUE
			   -DOPTION_USE_SYSTEM_ZLIB=TRUE	
			   -DOPTION_USE_SYSTEM_LIBPNG=TRUE	
			   -DOPTION_USE_SYSTEM_LIBJPEG=TRUE 		   
			   -DCMAKE_INSTALL_LIBDIR=<INSTALL_DIR>/lib	
	LOG_CONFIGURE 1
	LOG_BUILD 1
	LOG_INSTALL 1
)

add_custom_command(TARGET FLTK
	POST_BUILD
	WORKING_DIRECTORY 
		${CMAKE_INSTALL_PREFIX}/lib
	COMMAND ${CMAKE_COMMAND} -E create_symlink libfltk_SHARED.so libfltk.so
	COMMAND ${CMAKE_COMMAND} -E create_symlink libfltk_images_SHARED.so libfltk_images.so
	COMMAND ${CMAKE_COMMAND} -E create_symlink libfltk_forms_SHARED.so libfltk_forms.so
	COMMAND ${CMAKE_COMMAND} -E create_symlink libfltk_gl_SHARED.so libfltk_gl.so
	BYPRODUCTS 
		libfltk.so
		libfltk_images.so
		libfltk_forms.so
		libfltk_gl.so
	COMMENT "Creating symbolic links to SHARED fltk libraries..."
)

```

**Comment by Albrecht-S** [comment]
@KBentley57 Sorry for the late comment, but anyway:
1. Thanks for your comment, this can be very helpful but is slightly OT here since this issue is about `fltk-config` (and FLTK's [_maybe future_] `pkg-config` support). I appreciate your comment nevertheless
2. The issue with the `_SHARED` suffixes you're resolving with the code you mentioned above has been fixed in FLTK 1.4 (Git master, no release yet). If you or anybody else like to try it, the [latest snapshot](https://www.fltk.org/software.php?VERSION=1.4.x#developer) on our [download page](https://www.fltk.org/software.php) has these changes incorporated
3. You used 1.3.5 in your example code. I'm not sure if the changes mentioned above will make it into a 1.3(.6) release, but I'm thinking about backporting it to FLTK 1.3 (Git branch-1.3)

**Comment by Albrecht-S** [comment]
@KBentley57 Sorry for **my confusion**, I think now I got it. Your comment was absolute on topic because your CMake code generates a valid workaround for this issue. You probably know why, but for other readers, please see my next comment.

**Comment by Albrecht-S** [comment]
So, let me rephrase the problem: the OP @dvzrv said:
> The fltk-config tool should provide the path to libfltk.so to be able to build against the shared libraries.

The problem - as I see it now - is (was) that the shared libraries (at least those created by CMake builds) were not named properly: they had the suffix `_SHARED`, for instance `libfltk_SHARED.so` which is usually a symlink to `libfltk_SHARED.so.1.4` or similar.

The above mentioned change in FLTK 1.4 should fix this since shared libs don't have the _SHARED suffix anymore.

To be precise, here's the (stripped) output of `fltk-config` on my Linux system (FLTK 1.4):
```
$ fltk-config --use-images --ldstaticflags
/usr/local/fltk/lib/libfltk_images.a -ljpeg -lpng -lz /usr/local/fltk/lib/libfltk.a …

$ fltk-config --use-images --ldflags
-L/usr/local/fltk/lib -lfltk_images -ljpeg -lpng -lz -lfltk …
```
As you can see, the first call provides full paths to the static libs, whereas the second call provides the standard `-lfltk_images` and `-lfltk` linker flags used to link with the dynamic (shared) libs **if they exist** - or to the static libs if shared libs are not found.

That said, I believe this issue is resolved (can we close it), unless the OP @dvzrv meant that fltk-config should provide the **FULL PATH** to the shared libraries. Is that what you meant, or am I missing something? Please elaborate or close the issue. TIA.

**Comment by Albrecht-S** [comment]
If you close this issue and would still like to have `pkg-config` support in FLTK, please open another issue. Before you do so, please note that we do already have [STR 2180](https://www.fltk.org/str.php?L2180) that requests `pkg-config` support.

I believe that we'd need help from someone who knows `pkg-config` well so this person is able to provide `pkg-config` templates for FLTK which we could use to fill in the correct names or whatever. Patches or PR's would be appreciated. Something like those provided in STR 2180. Are these complete? Thanks in advance to consider this.

**Comment by Albrecht-S** [comment]
Note: I added label 'fixed' for now, but I'm willing to remove it if there are any new aspects. This issue will be closed in a few days otherwise.

**Comment by Albrecht-S** [comment]
Closing this issue now.
